code/unclassified/leap_year/leap_year.cpp

```diff
@@ -0,0 +1,30 @@
+// part of cosmos by opengenus foundation
+
+// THIS PROGRAM TAKES AN INPUT FOR YEAR AND RETURNS TRUE IF YEAR IS A LEAP YEAR
+
+
+#include <iostream>
+using namespace std;
+
+bool isLeap(int year)
+{
+   return (year % 4 == 0 && year % 100 != 0) || year % 400 == 0; 
+}
+
+int main()
+{
+   int year = 2000, year_2 = 3000;
+   cout<<"Leap years from "<<year<<" to "<<year_2<<endl;
+
+   while(year <= year_2)
+   {
+      if(isLeap(year))
+      {
+         cout<<year<<endl;
+      }
+
+      year++;
```

Comment:
Pre increment saves copy, although it makes no difference here. Also, why this ruins the point, if I were to get the leap years between two years, I would directly iterate over them, using increments of 4, and checking for 100s at various points. A better way to test his function would be creating tests with catch2, and/or getting a year from user input, and checking if it is a leap year.